### PR TITLE
support JSONpath queries

### PR DESCRIPTION
**Pull request by ryanSoley**
closes: #315 
---

## What
  * finally getting around to merging this integration branch to support JSONpath queries of logged rubicon data
    * take a look at the closed stores in [this milestone](https://github.com/capitalone/rubicon-ml/milestone/8?closed=1) for more context

## To Do
- [x] merge #384 for tests here to pass
- [x] update notebook example
- [x] update PR description and remove draft status

## How to Test
  * `python -m pytest`

### CODE REVIEW

File: rubicon_ml/cli.py [hunk 1]
```diff
@@ -78,7 +82,75 @@ def ui(root_dir, page_size, project_name, host, port, debug, storage_options):
...
+)
+@click.argument(
+    "query",
+    nargs=1,
+    required=True,
+)
+def search(root_dir, project_name, color, bf, query):
+    """
+    This function provides capability to query  rubicon experiments from the command line without using python
+    directly and output the JSON in the command line.
```
Comment (shania-m) [inline rubicon_ml/cli.py:127]: can we get doc strings for the parameters? Or `bf` specifically

File: rubicon_ml/cli.py [hunk 2]
```diff
@@ -78,7 +82,75 @@ def ui(root_dir, page_size, project_name, host, port, debug, storage_options):
...
+
+    rb_json = RubiconJSON(projects=project)
+    res = rb_json.search(query)
+    if not isinstance(res, list):
+        res = [res]
+    matches = [r.value for r in res]
+    if "color" in click.get_current_context().params:
+        fg_color = color
+
+    result = matches if bf else pprint.pformat(matches, indent=1)
```
Comment (shania-m) [inline rubicon_ml/cli.py:149]: what is `bf`?

**Review by stephenpardy — COMMENTED** [review]
We don't currently enforce type hints, but can we please add them here wherever possible to facilitate adding mypy in the future? Thanks!

File: rubicon_ml/cli.py (hunk 2 above)
Comment (ryanSoley) [reply rubicon_ml/cli.py:149]: that and the color option were for formatting the output. I replaced `bf` with `pp` for pretty-print, which is what `bf` was actually toggling. I also removed the color output cause we really don't need to support that

File: rubicon_ml/cli.py (hunk 1 above)
Comment (ryanSoley) [reply rubicon_ml/cli.py]: they're in the definitions of each of the `click.options` - you can view the full help message with `rubicon_ml search --help`

**Review by stephenpardy — APPROVED** [review]

**Pull request merged**